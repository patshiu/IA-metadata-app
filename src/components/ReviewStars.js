import PropTypes from 'prop-types';

const ReviewStars = ({ reviewStars }) => {
  if (reviewStars) {
    const numStars = Number(reviewStars);
    let starsIcon = '☆☆☆☆☆';
    for (let i = 0; i < numStars; i++) {
      starsIcon = '★' + starsIcon;
      starsIcon = starsIcon.slice(0, 5);
    }

    return <>{starsIcon}</>;
  } else {
    return null;
  }
};

ReviewStars.propTypes = {
  reviewStars: PropTypes.string
};

export default ReviewStars;
