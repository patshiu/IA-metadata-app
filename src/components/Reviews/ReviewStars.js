import React from 'react';
import PropTypes from 'prop-types';

const ReviewStars = ({ reviewStars }) => {
  if (reviewStars) {
    const numStars = Number(reviewStars);
    let starsIcon = '☆☆☆☆☆';
    for (let i = 0; i < numStars; i++) {
      starsIcon = '★' + starsIcon;
      starsIcon = starsIcon.slice(0, 5);
    }
    const aria = 'Rating: ' + reviewStars + ' of 5 stars';
    return <span aria-description={aria}>{starsIcon}</span>;
  } else {
    return null;
  }
};

ReviewStars.propTypes = {
  reviewStars: PropTypes.string
};

export default ReviewStars;
