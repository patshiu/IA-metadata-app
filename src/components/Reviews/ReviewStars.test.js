import React from 'react';
import renderer from 'react-test-renderer';

import ReviewStars from './ReviewStars';

test('takes rating as string and renders review stars with aria description', () => {
  const reviewStars = '4';
  const tree = renderer.create(<ReviewStars reviewStars={reviewStars} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('if rating is undefined, renders nothing', () => {
  const reviewStars = undefined;
  const tree = renderer.create(<ReviewStars reviewStars={reviewStars} />).toJSON();
  expect(tree).toMatchSnapshot();
});
