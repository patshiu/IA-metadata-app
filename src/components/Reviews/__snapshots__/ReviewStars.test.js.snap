// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if rating is undefined, renders nothing 1`] = `null`;

exports[`takes rating as string and renders review stars with aria description 1`] = `
<span
  aria-description="Rating: 4 of 5 stars"
>
  ★★★★☆
</span>
`;
